src/nvim/ops.c

```diff
@@ -6291,3 +6287,20 @@ bool op_reg_set_previous(const char name)
   y_previous = &y_regs[i];
   return true;
 }
+
+bcount_t get_region_bytecount(linenr_T start_lnum, linenr_T end_lnum,
+                              colnr_T start_col, colnr_T end_col)
+{
+  const char *first = (const char *)ml_get(start_lnum);
+  bcount_t deleted_bytes = (bcount_t)STRLEN(first) - start_col + 1;
+
+  if (start_lnum == end_lnum) {
+    return deleted_bytes - ((bcount_t)STRLEN(first) - end_col + 1);
+  }
+
+  for (linenr_T i = 1; i <= end_lnum-start_lnum-1; i++) {
+    deleted_bytes += (bcount_t)STRLEN(
+        ml_get(start_lnum + i)) + 1;
+  }
+  return deleted_bytes + end_col + 1;
```

Comment:
this should just be `end_col`, there is no extra newline here.

Reply:
I think I had the `end_col + 1` since the function was end-inclusive. I changed it to be end-exclusive instead, which also resolves the other comment you had.